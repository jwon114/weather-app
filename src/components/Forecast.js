import React from 'react';
import PropTypes from 'prop-types';

const Forecast = (props) => (
  <div className="forecast">
    <div>MON</div>
    <div>TUE</div>
    <div>WED</div>
    <div>THU</div>
    <div>FRI</div>
  </div>
);

Forecast.propTypes = {

};

export default Forecast;