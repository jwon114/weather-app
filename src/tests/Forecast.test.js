import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { render } from '@testing-library/react';
import Forecast from '../components/Forecast';

test('should render Forecast correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<Forecast />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('should render the the next 5 days', () => {
  const { container, getByText } = render(<Forecast data={['Mon', 'Tue', 'Wed', 'Thu', 'Fri']} />);
  expect(container.firstChild.children.length).toBe(5);
  expect(getByText(/mon/i)).toBeInTheDocument();
  expect(getByText(/tue/i)).toBeInTheDocument();
  expect(getByText(/wed/i)).toBeInTheDocument();
  expect(getByText(/thu/i)).toBeInTheDocument();
  expect(getByText(/fri/i)).toBeInTheDocument();
});
