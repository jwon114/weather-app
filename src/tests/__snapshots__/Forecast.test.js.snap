// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Forecast correctly 1`] = `
<div
  className="forecast"
>
  <div>
    MON
  </div>
  <div>
    TUE
  </div>
  <div>
    WED
  </div>
  <div>
    THU
  </div>
  <div>
    FRI
  </div>
</div>
`;
